Sources/UnleashProxyClientSwift/Metrics/Metrics.swift

```diff
@@ -34,21 +37,36 @@ public class Metrics {
         self.bucket = Bucket(clock: clock)
         self.customHeaders = customHeaders
         self.connectionId = connectionId
+        self.queue = queue
     }
 
     func start() {
         if disableMetrics { return }
 
-        self.timer = Timer.scheduledTimer(withTimeInterval: metricsInterval, repeats: true) { _ in
+        queue.sync {
+            self.timer?.cancel()
+            self.timer = nil
+        }
+
+        let timer = DispatchSource.makeTimerSource(queue: DispatchQueue.global(qos: .background))
```

Comment:
Using DispatchSourceTimer for thread-safe, run-loop-independent, and precise background execution